apps/zotonic_core/src/models/m_identity.erl

```diff
@@ -1477,27 +1477,68 @@ insert_unique(RscId, Type, Key, Props, Context) ->
 
 
 %% @doc Set the visited timestamp for the given user.
-%% @todo Make this a log - so that we can see the last visits and check if this
-%% is from a new browser/ip address.
 -spec set_visited(m_rsc:resource_id(), z:context()) -> ok | {error, enoent}.
-set_visited(undefined, _Context) ->
-    ok;
+set_visited(undefined, _Context) -> ok;
 set_visited(UserId, Context) when is_integer(UserId) ->
-    case z_db:q(
-        "update identity
-         set visited = now()
-         where rsc_id = $1
-           and type = 'username_pw'",
-        [m_rsc:rid(UserId, Context)],
-        Context)
-    of
-        0 ->
-            {error, enoent};
-        N when N >= 1 ->
-            flush(UserId, Context),
-            ok
+    z_db:transaction(fun(Ctx) -> set_visited_trans(UserId, Ctx) end, Context);
+set_visited(User, Context) ->
+    set_visited(m_rsc:rid(User, Context), Context).
+
+% Part of 'set_visited' to execute in one DB transaction
+set_visited_trans(UserId, Context) when is_integer(UserId) ->
+    case z_db:q1("select id from identity where rsc_id = $1 and type = 'username_pw'", [UserId], Context) of
+        undefined -> {error, enoent};
+        EntryId ->
+            UserAgent = z_convert:to_binary(m_req:get(user_agent, Context)),
+            IpAddress = z_convert:to_binary(m_req:get(peer, Context)),
```

Comment:
Maybe use `inet_parse:ntoa/1` (and then iolist_to_binary).
Of course will need a check for the case where the peer is `undefined`.

Reply:
Done in 6d385cf382a3afe72a2281c5f24ca3e16a8edc42